modules/examiner/php/NDB_Menu_Filter_Form_examiner.class.inc

```diff
@@ -0,0 +1,327 @@
+
+/**
+ * Examiner module menu filter form
+ *
+ * PHP Version 5
+ *
+ * @category Clinical
+ * @package  Loris
+ * @author   Tara Campbell <tara.campbell@mail.mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris
+ */
+
+class NDB_Menu_Filter_Form_Examiner extends NDB_Menu_Filter_Form
+{
+    /**
+     * Checks if the user has access to examiner menu filter
+     *
+     * @return boolean
+     */
+    function _hasAccess()
+    {
+        $user = User::singleton();
+        return $user->hasPermission('examiner_view');
+    }
+
+    /**
+     * Sets the menu filter class variables.
+     *
+     * @return boolean
+     */
+    function _setupVariables()
+    {
+        $config           = NDB_Config::singleton();
+        $certification    = $config->getSetting('Certification');
+        $useCertification = $certification['EnableCertification'];
+
+        // base queries
+        $query     = " FROM examiners e LEFT JOIN psc ON (e.centerID=psc.CenterID)";
+        $certQuery = " LEFT JOIN certification c ON (c.examinerID = e.examinerID) 
+            LEFT JOIN test_names tn ON (c.testID = tn.ID)";
+
+        if ($useCertification == '1') {
+            $query .= $certQuery;
+        }
+
+        $query .= ' WHERE 1=1';
+
+        $user = User::singleton();
+        if (!$user->hasPermission('examiner_multisite')) {
+            $query .= " AND e.CenterID=" . $user->getData('CenterID');
+        }
+
+        // set the class variables
+        $this->columns      = array(
+                               'e.full_name as Examiner',
+                               'psc.Name as Site',
+                               'e.radiologist as Radiologist',
+                              );
+        $this->query        = $query;
+        $this->order_by     = 'e.full_name';
+        $this->headers      = array(
+                               'Examiner',
+                               'Site',
+                               'Radiologist',
+                              );
+        $this->validFilters = array(
+                               'e.full_name',
+                               'e.centerID',
+                               'COALESCE(e.radiologist, 0)',
+                              );
+
+        $this->formToFilter = array(
+                               'examiner'    => 'e.full_name',
+                               'site'        => 'e.centerID',
+                               'radiologist' => 'COALESCE(e.radiologist, 0)',
+                              );
+
+        // If certification is turned on, update the class variables to include
+        // certification information
+        if ($useCertification == '1') {
+            array_push(
+                $this->columns,
+                'tn.Full_name as Certification',
+                'e.examinerID as ID',
+                'c.pass as Pass'
+            );
+            array_push($this->headers, 'Certification');
```

Comment:
according to the array_push documentation "Note: If you use array_push() to add one element to the array it's better to use $array[] ="

(If you change that for these, you should also probably be consistent and do it for $this->columns too)